### PR TITLE
Z index for dropdown selector collide with modals

### PR DESCRIPTION
**Pull request by Afsoon**
#### Summary
Fixing a z-index issue with dropdown selector and modal with z-index. Ideally we should use this property at input level, instead of form, but the fix it's much bigger and require more investigation, as the z index levels are complex to follow. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-47802

#### Screenshots
before
AlwaysOnTop

After
bug_fixed

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/admin_console/billing/company_info_edit.scss [hunk 1]
```diff
@@ -5,6 +5,7 @@
     background-color: var(--sys-center-channel-bg);
     border-radius: 4px;
     box-shadow: 0 2px 3px rgba(0, 0, 0, 0.08);
+    isolation: isolate;
```
Comment (Afsoon) [inline components/admin_console/billing/company_info_edit.scss:8]: Why the fix is one property? As I say on the summary ideally should be at level component but I'm having some issue following the z-index between inputs that would make the fix much bigger. This property is a hidden gem of CSS, and it isn't new, that help us to make it reasonable to manage z-index values. This property does one thing and it is create a new stacking context, more info on this [post from Josh w Comeau](https://www.joshwcomeau.com/css/stacking-contexts/#airtight-abstractions-with-isolation).

**Review by AGMETEOR — APPROVED** [review]
If it solves the problem and introduces no regressions then it's indeed a hidden gem

File: components/admin_console/billing/company_info_edit.scss (hunk 1 above)
Comment (neallred) [reply components/admin_console/billing/company_info_edit.scss:8]: Awesome post! Didn't know about `isolation`, either. I can vouch for the `z-context` plugin's awesomeness, but haven't tried the other.

**Review by neallred — APPROVED** [review]

**Review by stevemudie — APPROVED** [review]
Looks good @Afsoon! Thanks for the quick fix! 🎉

**Pull request merged**